rs/crypto/internal/crypto_lib/bls12_381/type/src/interpolation.rs

```diff
@@ -1,30 +1,71 @@
 use crate::{G1Affine, G1Projective, G2Affine, G2Projective, NodeIndex, Scalar};
 
 #[derive(Copy, Clone, Debug)]
-/// Error during interpolation
-pub enum InterpolationError {
-    /// The coefficients were empty
-    NoCoefficients,
+/// Error when constructing NodeIndices
+pub enum InvalidNodeIndices {
     /// A node index was duplicated
     DuplicatedNodeIndex,
+}
+
+#[derive(Copy, Clone, Debug)]
+/// Error during interpolation
+pub enum InterpolationError {
     /// No samples were provided for interpolation
     NoSamples,
     /// The set of samples did not match the coefficient count
     WrongSampleCount,
 }
 
+/// Lagrange interoplation samples
+pub struct NodeIndices {
+    index: Vec<NodeIndex>,
```

Comment:
nit: how about renaming to `indices`, given that `index` is singular and this is a `Vec`. Or was the singular chosen deliberately?

Reply:
:+1:

Reply:
Done